### DISCUSSION TITLE
Perfrom LocalSearch withouth combining with embeding store.

### DISCUSSION THREAD

**Discussion by jorgelunams** [microsoft/graphrag#863] [General]
How can we run LocalSearch withouth 

entity_df = pd.read_parquet(f"{INPUT_DIR}/{ENTITY_TABLE}.parquet")
entity_embedding_df = pd.read_parquet(f"{INPUT_DIR}/{ENTITY_EMBEDDING_TABLE}.parquet")

entities = read_indexer_entities(entity_df, entity_embedding_df, COMMUNITY_LEVEL)

# load description embeddings to an in-memory lancedb vectorstore
# to connect to a remote db, specify url and port values.
description_embedding_store = LanceDBVectorStore(
    collection_name="entity_description_embeddings",
)
description_embedding_store.connect(db_uri=LANCEDB_URI)
entity_description_embeddings = store_entity_semantic_embeddings(
    entities=entities, vectorstore=description_embedding_store
)
print(LANCEDB_URI)
print(f"Entity count: {len(entity_df)}")
entity_df.head()

In our case we haave only the Knowledge Graph and not access to a Vector DB like the search sample for operation dulce.

**Comment by natoverse** [comment]
You could do this with a bit of work. The CLI can be a helpful example of how to load and execute a search engine. In [this code](https://github.com/microsoft/graphrag/blob/c451aa0093757faaab3f0bdb351d6a73a96e65f2/graphrag/query/cli.py#L124), we invoke a [factory function](https://github.com/microsoft/graphrag/blob/c451aa0093757faaab3f0bdb351d6a73a96e65f2/graphrag/query/factories.py#L102) that creates an instance of the [LocalSearch class](https://github.com/microsoft/graphrag/blob/c451aa0093757faaab3f0bdb351d6a73a96e65f2/graphrag/query/structured_search/local_search/search.py#L31). This factory function accepts a `description_embedding_store` parameter which needs to be an implementation of [BaseVectorStore](https://github.com/microsoft/graphrag/blob/c451aa0093757faaab3f0bdb351d6a73a96e65f2/graphrag/vector_stores/base.py#L40). We supply implementations for Azure AI Search and LanceDB in GraphRAG, but you could write your own that does whatever lookup you need to supply the entity matches.
